src/npm.rs

```diff
@@ -640,26 +632,15 @@ impl NpmPackageReqReference {
     specifier: &str,
   ) -> Result<Self, NpmPackageReqReferenceParseError> {
     let original_text = specifier;
-    let specifier = match specifier.strip_prefix("npm:") {
-      Some(s) => {
-        // Strip leading slash, which might come from import map
-        s.strip_prefix('/').unwrap_or(s)
-      }
+    let input = match specifier.strip_prefix("npm:") {
+      Some(input) => input,
       None => {
-        // don't allocate a string here and instead use a static string
-        // because this is hit a lot when a url is not an npm specifier
+        // this is hit a lot when a url is not an npm specifier
+        // so ensure nothing heavy occurs before this
         return Err(NpmPackageReqReferenceParseError::NotNpmSpecifier);
       }
     };
-    let parts = specifier.split('/').collect::<Vec<_>>();
```

Comment:
I changed the code to allocate a lot less. I didn't do any performance profiling on the improvement though.